src/test/ui/consts/const-mut-refs/const_mut_refs.rs

```diff
@@ -0,0 +1,17 @@
+// run-pass
+
+#![feature(const_mut_refs)]
+
+struct Foo {
+    x: usize
+}
+
+const fn bar(foo: &mut Foo) -> usize {
+    let x = &mut foo.x;
+    *x = 1;
```

Comment:
The previous test (directly assigning to a field of a reference to a struct ) was also a good case to test. Please reintroduce it